lib/object/cache.rb

```diff
@@ -44,11 +44,11 @@ class << self
     #   Cache.new { item } # item is only stored once, and then always
     #                      # retrieved, even if it is a different item
     #
-    def new(key = nil, ttl: default_ttl, key_prefix: default_key_prefix)
+    def new(key = nil, ttl: default_ttl, key_prefix: default_key_prefix, &block)
       return yield unless replica
 
       begin
-        key = build_key(key, key_prefix, Proc.new)
+        key = build_key(key, key_prefix, block)
```

Comment:
I think the reasoning was that somehow the caller location was used to generate the key, while you can use the block, you should also be able to use `binding` which makes more sense to me.

It could be that we need to change line 109.

Reply:
I tried refactoring it to binding but tests where multiple instances of the same type are cached are failing. I'm running into the limits of my Ruby knowledge here, could you perhaps have a look?